scripts/04-effect-for-beginners.md

```diff
@@ -0,0 +1,1622 @@
+---
+theme: blood
+highlightTheme: css/vs2015.css
+---
+# Effect for Beginners
+![[effect-logo.png]]
+### A guide through Effect's core concepts
+
+notes:
+Effect is a typescript library that gives you the tools to make it easy to write complex applications in a safe and composable manner. This video is a detailed introductory guide to the core effect concepts.
+
+We'll start with the basics and then move on to slowly build an example program piece by piece, which by the end will feature concurrent asynchronous work, granular error handling, dependency injection and benchmarking.
+
+If you are still curious about why Effect is worth learning, feel free check out my other videos on Effect linked in the cards now, and come back after.
+
+---
+
+Sponsored By: \
+The Effect Team
+
+notes:
+Effect is an amazing library and I want to quickly thank the Effect team for financially supporting the creation of content around it. Now on to the tutorial!
+
+---
+```
+npm install effect
+yarn add effect
+pnpm add effect
+```
+```ts
+import { Effect } from "effect"
+```
+notes:
+First, make sure to add the effect package to your project using your package manager of choice
+with that out of the way lets get to business
+
+---
+
+```ts
+type Effect<Requirements, Error, Value> = (
+  requirements: Requirements
+) => Value| Error;
+```
+
+notes:
+We'll begin with the Effect type. This type is the center of the enitre Effect ecosystem.
+
+The Effect type is generic over 3 parameters which describe its behavior. Requirements, Error, and Value.
+
+You should conceptualize an Effect as a function that takes in its requirements as an argument and returns either a value or an error.
+
+Its important to emphasize this function-like nature of an Effect. Just like a function, an Effect is only a description of its behavior when run. The output value is yielded only when the Effect is ran.
+
+---
+
+The unit `Effect`
+```ts
+type Unit = Effect<never, never, void>;
+```
+
+notes:
+The unit effect is the most basic Effect type. It describes a program that has requirements of type never, meaning it does not have any dependencies. It has a error also of type never, meaning it will never error, and it has a value of type void, meaning it does not return anything.
```

Comment:
`anything => anything meaningful`

Maybe worth to add a plain function example to continue the parallel:

```ts
const unit = (): void => {}
```